source-code/core/tsconfig.json

```diff
@@ -5,7 +5,8 @@
 		// DOM because of fetch. TODO: extract fetch from this package and
 		// let fetch be provided by the environment.
 		"lib": ["ESNext", "DOM"],
-		"types": [],
+		// need to specify node because `$fs` has a node type dependency
+		"types": ["node"],
```

Comment:
what was the issue that you needed to add this?

Reply:
$fs uses buffer to use node's filesystem. this "bug" has likely been introduced by https://github.com/inlang/inlang/issues/391. 

We need to build our own filesystem implementation with node independent types. I didn't want to change fs related things in this PR though.

FYI the build step fails otherwise

Reply:
You could also import it from "node:fs" and add the `eslint-ignore` on top. Then it is more clear that something needs to change. With global types it is hard to see where we use node types.

Reply:
Good hint. node does not export buffer though. I hope I fixed the issue with:

```ts
// eslint-disable-next-line no-restricted-imports
import type fs from "node:fs/promises"

/**
 * Minimal filesystem required by inlang to work
 *
 * TODO: node independent fs
 */
export type FS = {
	readFile: (
		...args: [id: string] | Parameters<fs.FileHandle["readFile"]>
	) => Promise<string> | ReturnType<fs.FileHandle["readFile"]>
	writeFile: (file: string, data: string) => Promise<void>
	readdir: (path: string) => Promise<Array<string | object>>
}
```

I expect type errors down the road but we need to build our own fs anyways.